enterprise/server/test/integration/remote_execution/remote_execution_test.go

```diff
@@ -35,6 +36,68 @@ func TestSimpleCommandWithNonZeroExitCode(t *testing.T) {
 	assert.Equal(t, "bye\n", res.Stderr, "stderr should be propagated")
 }
 
+func TestActionResultCacheWithSuccessfulAction(t *testing.T) {
+	rbe := rbetest.NewRBETestEnv(t)
+
+	rbe.AddBuildBuddyServer()
+	rbe.AddExecutor()
+
+	platform := &repb.Platform{
+		Properties: []*repb.Platform_Property{
+			{Name: "OSFamily", Value: runtime.GOOS},
+			{Name: "Arch", Value: runtime.GOARCH},
+		},
+	}
+	invocationID := "testabc123"
+	cmd := rbe.Execute(&repb.Command{
+		Arguments: []string{"echo"},
+		Platform:  platform,
+	}, &rbetest.ExecuteOpts{
+		InvocationID: invocationID,
+	})
+
+	res := cmd.Wait()
+	assert.Equal(t, 0, res.ExitCode, "exit code should be propagated")
+
+	_, err := rbe.GetActionResultForFailedAction(context.Background(), cmd, invocationID)
+	assert.True(t, status.IsNotFoundError(err))
+}
+
+func TestActionResultCacheWithFailedAction(t *testing.T) {
+	rbe := rbetest.NewRBETestEnv(t)
+
+	rbe.AddBuildBuddyServer()
+	rbe.AddExecutor()
+
+	platform := &repb.Platform{
+		Properties: []*repb.Platform_Property{
+			{Name: "OSFamily", Value: runtime.GOOS},
+			{Name: "Arch", Value: runtime.GOARCH},
+		},
+	}
+	invocationID := "testabc123"
+	cmd := rbe.Execute(&repb.Command{
+		Arguments: []string{"sh", "-c", "echo hello && echo bye >&2 && exit 5"},
+		Platform:  platform,
+	}, &rbetest.ExecuteOpts{
+		InvocationID: invocationID,
+	})
+
+	res := cmd.Wait()
+	assert.Equal(t, 5, res.ExitCode, "exit code should be propagated")
```

Comment:
nit: consider adding an assertion just after this line that looking up the ActionResult from the normal AC returns NotFound?

Reply:
Done